src/Classifier.lhs

```diff
@@ -44,6 +46,33 @@ showAuto alaws = "   1. simps   -> "   ++ showSimps (simps alaws)  ++ "\n"
               ++ "   2. folds   -> "   ++ concat (folds alaws)     ++ "\n"
               ++ "   3. unfolds -> "   ++ concat (unfolds alaws)    ++ "\n"
 
-addLawsClassifier :: Term -> AutoLaws -> AutoLaws
-addLawsClassifier te au = au
+addLawsClassifier :: NmdAssertion -> AutoLaws -> AutoLaws
+addLawsClassifier (nme, asser) au = AutoLaws {  simps = simps au ++ addSimp nme (assnT asser)
+                                             ,  folds = folds au
+                                             ,  unfolds = unfolds au
+                                             }
+
+termSize :: Term -> Int
+termSize (Val _ _)            =  1
+termSize (Var _ _)            =  1
+termSize (Cons _ _ _ ts)      =  1 + sum (map termSize ts)
+termSize (Bnd _ _ vs t)       =  1 + S.size vs + termSize t
+termSize (Lam _ _ vl t)       =  1 + length vl + termSize t
+termSize (Cls _ t)            =  1 + termSize t
+termSize (Sub _ t subs)       =  1 + termSize t + subsSize subs
+termSize (Iter _ _ _ _ _ vl)  =  3 + length vl
+termSize (Typ _)              =  2
+
+subsSize (Substn ts lvs)      =  3 * S.size ts + 2 * S.size lvs
+
+addSimp :: String -> Term -> [(String, Direction)]
+addSimp nme (PCons sb n (p:q:xs)) = do
```

Comment:
I added a comment here that then got deleted!!!!

You need to pattern match for a list of length 2 only - and to chec that the name (`n`) matches that for equivalence.  The equivalence Identifier is obtained from the `theEqv` field of the `LogicSig` component of `REqState`.

Reply:
Do i need to include logicSig in the method?

Reply:
Alright, I have committed changes that should pattern match more specifically for list of length 2 and for equivalence